src/main/java/frc/robot/commands/ShooterCommand.java

```diff
@@ -1,42 +1,105 @@
 package frc.robot.commands;
 
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+import frc.robot.Constants;
 import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Shooter;
 import frc.robot.subsystems.Vision;
+import frc.robot.subsystems.Shooter.ShooterSpeeds;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
-
-public class ShooterCommand extends CommandBase{
+public class ShooterCommand extends CommandBase {
+    /**
+     * Creates a new ShooterCommand.
+     */
     Shooter m_shooter;
     Intake m_intake;
     Vision m_vision;
 
+    double m_distance;
+
+    LigerTimer m_shootDelay = new LigerTimer(Constants.SHOOTER_MOTOR_WAIT_TIME);
+    LigerTimer m_intakeDelay = new LigerTimer(Constants.SHOOTER_INTAKE_WAIT_TIME);
+    LigerTimer m_shotTime = new LigerTimer(Constants.SHOTS_WAIT_TIME);
+    LigerTimer m_intakeTime = new LigerTimer(Constants.INTAKE_WAIT_TIME);
+
+    ShooterSpeeds m_shooterSpeeds;
+
+    enum State {
+        FINDING_VISION_TARGET, SPEED_UP_SHOOTER, WAIT_FOR_SHOOTER, TURN_ON_CHUTE, TURN_ON_INTAKE, WAIT_FOR_SHOTS, WAIT_FOR_INTAKE;
+    }
+
+    State m_state;
+
     public ShooterCommand(Shooter shooter, Intake intake, Vision vision) {
         m_shooter = shooter;
         m_intake = intake;
         m_vision = vision;
     }
 
+    // Called when the command is initially scheduled.
     @Override
     public void initialize() {
-        System.out.println("*sound of shooter shooting*"); // debugging
+        m_state = State.SPEED_UP_SHOOTER;
     }
+
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
-        
+        switch (m_state) {
+            case FINDING_VISION_TARGET:
+                m_distance = m_vision.getDistance();
+                if(m_distance != 0.0)
+                    m_state = State.SPEED_UP_SHOOTER;
+                break;
```

Comment:
This should be a fall-through. make sure to include a comment.

Reply:
Unfortunately, I think we want a timeout on this state. The vision might not be working and we don't want to wait forever. I think you can use a LigerTimer with maybe 2 seconds, but it does not need to be a Constants number. If it times out, use 9 ft for the distance (should be local constant in the class).